Tests/Unit/NetworkServer/CupsUpdateInfoResponseHeaderTests.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.Tests.Unit.NetworkServer
+{
+    using System;
+    using System.Buffers;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using LoRaWan.NetworkServer.BasicsStation;
+    using Xunit;
+
+    public class CupsUpdateInfoResponseHeaderTests
+    {
+        private readonly string updateUriString = "https://localhost:1234";
+
+        [Fact]
+        public void Serialize_WithNoUpdates()
+        {
+            var updateResponseHeader = new CupsUpdateInfoResponseHeader
+            {
+                SignatureKeyCrc = 0,
+                UpdateDataLength = 0,
+            };
+
+            using var memoryPool = MemoryPool<byte>.Shared.Rent(2048);
+
+            // Act
+            var serializedResponse = updateResponseHeader.Serialize(memoryPool.Memory.Span);
+
+            // Assert
+            Assert.True(serializedResponse.ToArray().All(b => b == 0));
+        }
+
+        [Fact]
+        public void Serialize_WithFirmwareUpdates()
+        {
+            // setting up the twin in such a way that there are only firmware updates
+            var signatureBytes = Convert.FromBase64String("ABCD");
+            var keyCRC = 6789U;
+            var updateDataLength = 100U;
```

Comment:
These can be constants:

```suggestion
            const uint keyCRC = 6789;
            const uint updateDataLength = 100;
```